src/main/java/com/airsquared/blobsaver/Controller.java

```diff
@@ -193,17 +234,23 @@ public void initialize() {
 
         // use macos menu bar or not
         if (PlatformUtil.isMac()) {
+            // use system menubar instead
+            menuBar.setUseSystemMenuBar(true);
+
+            // makes the app taller to compensate for the missing menu bar
+            ((VBox) menuBar.getParent()).setMinHeight(560.0);
+            ((VBox) menuBar.getParent()).setPrefHeight(560.0);
+            presetVBox.setMinHeight(560.0);
+            presetVBox.setPrefHeight(560.0);
             primaryStage.setOnShowing(new EventHandler<WindowEvent>() {
                 @Override
                 public void handle(WindowEvent event) {
-                    useMacOSMenuBar();
+                    Platform.runLater(() -> useMacOSMenuBar());
```

Comment:
Why are you doing `Platform.runLater()` here?

Reply:
Because the macOS menu bar sometimes gets generated before the stage is completely shown, so the menu bar contents are all messed up (this took a long time to figure out)

Reply:
Oh, ok.